packages/components/autocomplete/src/autocomplete.vue

```diff
@@ -0,0 +1,313 @@
+        </template>
+        <template v-if="$slots.suffix" #suffix>
+          <slot name="suffix"></slot>
+        </template>
+      </el-input>
+    </div>
+    <template #content>
+      <div
+        ref="regionRef"
+        :class="[ns.b('suggestion'), ns.is('loading', suggestionLoading)]"
+        :style="{ minWidth: dropdownWidth, outline: 'none' }"
+        role="region"
+      >
+        <el-scrollbar
+          :id="id"
+          tag="ul"
+          :wrap-class="ns.be('suggestion', 'wrap')"
+          :view-class="ns.be('suggestion', 'list')"
+          role="listbox"
+        >
+          <li v-if="suggestionLoading">
+            <el-icon :class="ns.is('loading')"><Loading /></el-icon>
+          </li>
+          <template v-else>
+            <li
+              v-for="(item, index) in suggestions"
+              :id="`${id}-item-${index}`"
+              :key="index"
+              :class="{ highlighted: highlightedIndex === index }"
+              role="option"
+              :aria-selected="highlightedIndex === index"
+              @click="handleSelect(item)"
+            >
+              <slot :item="item">{{ item[valueKey] }}</slot>
+            </li>
+          </template>
+        </el-scrollbar>
+      </div>
+    </template>
+  </el-tooltip>
+</template>
+
+<script lang="ts" setup>
+import {
+  ref,
+  computed,
+  onMounted,
+  nextTick,
+  useAttrs as useCompAttrs,
+} from 'vue'
+import { debounce } from 'lodash-unified'
+import { onClickOutside } from '@vueuse/core'
+import { useAttrs, useNamespace } from '@element-plus/hooks'
+import { generateId, isArray, throwError } from '@element-plus/utils'
+import { UPDATE_MODEL_EVENT } from '@element-plus/constants'
+import ElInput from '@element-plus/components/input'
+import ElScrollbar from '@element-plus/components/scrollbar'
+import ElTooltip from '@element-plus/components/tooltip'
+import { useDeprecateAppendToBody } from '@element-plus/components/popper'
+import ElIcon from '@element-plus/components/icon'
+import { Loading } from '@element-plus/icons-vue'
+import { autocompleteProps, autocompleteEmits } from './autocomplete'
+import type { StyleValue } from 'vue'
+
+defineOptions({
+  name: 'ElAutocomplete',
+  inheritAttrs: false,
+})
+
+const COMPONENT_NAME = 'ElAutocomplete'
+
+const props = defineProps(autocompleteProps)
+const emit = defineEmits(autocompleteEmits)
+
+const ns = useNamespace('autocomplete')
+const { compatTeleported } = useDeprecateAppendToBody(
+  COMPONENT_NAME,
+  'popperAppendToBody'
+)
+const attrs = useAttrs()
+const compAttrs = useCompAttrs()
+const suggestions = ref<any[]>([])
+const highlightedIndex = ref(-1)
+const dropdownWidth = ref('')
+const activated = ref(false)
+const suggestionDisabled = ref(false)
+const loading = ref(false)
+const inputRef = ref<{
+  inputOrTextarea: HTMLInputElement | HTMLTextAreaElement
+  focus: () => void
+  $el: HTMLElement
+} | null>(null)
+const regionRef = ref<HTMLElement | undefined>()
```

Comment:
```suggestion
const regionRef = ref<HTMLElement>()
```

Vue will add `undefined` automatically.